Bug: E2E Tests failing - missing @st44/types build step

## Description

E2E Tests workflow is failing on the **Build backend** step because the `@st44/types` package is not being built before attempting to build the backend. The backend has a dependency on `@st44/types`, but the workflow doesn't include a step to build this shared types package.

## Reproduction Steps

1. Workflow runs via schedule (daily at 2 AM UTC) or manual trigger
2. Workflow installs frontend dependencies
3. Workflow installs backend dependencies
4. Workflow attempts to build backend
5. Build fails with TypeScript errors

## Error Output

From run #20547734128 (2025-12-28):

```
❌ Cannot find module '@st44/types' or its corresponding type declarations
❌ Cannot find module '@st44/types/generators' or its corresponding type declarations
❌ Argument of type 'unknown' is not assignable to parameter of type 'ZodError<unknown>'
❌ 'validatedData' is of type 'unknown'
```

## Expected Behavior

The E2E workflow should:
1. Install root dependencies (or types package dependencies)
2. Build `packages/types` package
3. Install and build backend (which can now find `@st44/types`)
4. Run E2E tests successfully

## Root Cause

The workflow in `.github/workflows/e2e.yml` is missing the types build step:

**Current workflow (lines 44-54):**
```yaml
- name: Install frontend dependencies
  working-directory: apps/frontend
  run: npm ci

- name: Install backend dependencies
  working-directory: apps/backend
  run: npm ci

- name: Build backend  # ❌ Fails here
  working-directory: apps/backend
  run: npm run build
```

**Missing step:**
The workflow needs to build `@st44/types` before building the backend.

## Suggested Fix

Add a step to build the types package before building the backend:

```yaml
- name: Install types package dependencies
  working-directory: packages/types
  run: npm ci

- name: Build types package
  working-directory: packages/types
  run: npm run build

- name: Install backend dependencies
  working-directory: apps/backend
  run: npm ci

- name: Build backend
  working-directory: apps/backend
  run: npm run build
```

## Impact

- **Severity**: `critical` - Blocks all E2E testing in CI
- **Affected Workflow**: `.github/workflows/e2e.yml`
- **Last Successful Run**: Unknown (need to check)
- **Failing Since**: At least 2025-12-28 02:45:43

## Related Files

- `.github/workflows/e2e.yml` (lines 44-54)
- `packages/types/package.json`
- `apps/backend/package.json` (depends on `@st44/types`)

## Acceptance Criteria

- [ ] E2E workflow includes types package build step
- [ ] Types package is built before backend build
- [ ] Backend build succeeds in CI
- [ ] E2E tests can run successfully
- [ ] Workflow tested with manual trigger

## Reference

- Failed run: https://github.com/tidemann/st44-home/actions/runs/20547734128
- Error occurred in step: "Build backend"

--- a/.github/workflows/e2e.yml
+++ b/.github/workflows/e2e.yml
@@ -34,12 +34,21 @@ jobs:
           node-version: '22'
           cache: 'npm'
           cache-dependency-path: |
+            packages/types/package-lock.json
             apps/frontend/package-lock.json
             apps/backend/package-lock.json
 
       - name: Install Playwright Browsers
         working-directory: apps/frontend
         run: npx playwright install --with-deps chromium
+
+      - name: Install types package dependencies
+        working-directory: packages/types
+        run: npm ci
+
+      - name: Build types package
+        working-directory: packages/types
+        run: npm run build
 
       - name: Install frontend dependencies
         working-directory: apps/frontend

